nixos/modules/system/boot/systemd/initrd.nix

```diff
@@ -211,7 +216,7 @@ in {
     };
 
     root = lib.mkOption {
-      type = lib.types.enum [ "fstab" "gpt-auto" ];
+      type = lib.types.enum [ "fstab" "gpt-auto" "verity" ];
```

Comment:
I don't believe this is the right abstraction. You still need an `fstab` or `gpt-auto` for this to work because `systemd-veritysetup-generator` only opens `dm` devices. It doesn't mount anything and indeed you could configure your mounts with either the fstab or gpt-auto when you use dm-verity.

Reply:
It does mount stuff (see `roothash` in https://www.freedesktop.org/software/systemd/man/latest/systemd-veritysetup-generator.html), and it's incompatible with both `root=gpt-auto` and `root=fstab` unfortunately. I'd very much liked the latter to not be the case, but it is. 

Re "automatically mounting", also see [this Matrix discussion](https://matrix.to/#/!DBFhtjpqmJNENpLDOv:nixos.org/$WfP271DEhtCF47-AzBuos4kitpTA8IeO9r1HMHkOHls?via=nixos.org&via=matrix.org&via=tchncs.de).